### ISSUE TITLE
dxDrawLine3D, make possible nice connection between each line

### ISSUE THREAD

**Issue by CrosRoad95** [multitheftauto/mtasa-blue#871]
**Is your feature request related to a problem? Please describe.**
When you want to make 2 3d lines, nicely connected together, you just can't do this
![image](https://user-images.githubusercontent.com/22455534/56210898-0abb9e80-6057-11e9-8472-d2100de67f2d.png)
![image](https://user-images.githubusercontent.com/22455534/56210908-0e4f2580-6057-11e9-9dd9-bd7f8781f49b.png)
```lua
addEventHandler("onClientRender", root, function()
    dxDrawLine3D(-2442.79,731.83,35.02, -2432.79,731.83,35.02, tocolor(255,0,0), 20)
    dxDrawLine3D(-2442.79,731.83,35.02, -2442.79,721.83,35.02, tocolor(255,0,0), 20)
end)
``` 

**Describe the solution you'd like**
my solution is:
skip first `start x,y,z` in second dxDrawLine3D, and new line will start from last `end x,y,z` and gap will filed with another line 3d
![image](https://user-images.githubusercontent.com/22455534/56211988-6555fa00-6059-11e9-92e3-6a6a1662d102.png)
blue rectangle is new one which would be created

**Describe alternatives you've considered**
/

**Additional context**
This is only cosmetic issue

**Comment by Woovie** [comment]
I feel like this should be left to the persons executing the draw, you know the width and end position of each line. It's not too hard to calculate from there either a square that goes over the corner, or adjust the end points so that they extend to the edge of each other.

**Comment by dmi7ry** [comment]
As an option it may be a filled circle (it's simpler to calculate, because you have not needed to find the corners, etc)

**Comment by WADmitry** [comment]
As an option it may be a filled circle (it's simpler to calculate, because you have not needed to find the corners, etc)

**Comment by qaisjp** [comment]
Is a dxDrawCircle at the intersection point with `radius=lineWidth` suitable?

Also, love2d has [`love.graphics.setLineJoin`](https://love2d.org/wiki/love.graphics.setLineJoin) / [`LineJoin`](https://love2d.org/wiki/LineJoin) which is nice.

![image](https://user-images.githubusercontent.com/923242/56231289-0652ae00-6076-11e9-8826-eaa711903a39.png)


This is [their implementation](https://bitbucket.org/rude/love/src/32d137a38b56c326fe4ff2e1349de22088385632/src/modules/graphics/Polyline.cpp?at=default&fileviewer=file-view-default).

**Comment by WADmitry** [comment]
> Is a dxDrawCircle at the intersection point with radius=lineWidth suitable?

half of the lineWidth I guess

**Comment by qaisjp** [comment]
> I feel like this should be left to the persons executing the draw, you know the width and end position of each line. It's not too hard to calculate from there either a square that goes over the corner, or adjust the end points so that they extend to the edge of each other.

I agree, although we could probably publish this calculation as a "useful script" somewhere (as this offset changes depending on the angle between the two lines).

(Actually I don't think an offset is all you need — the `miter` and `bevel` options above are not possible with just offsets)

> > Is a dxDrawCircle at the intersection point with radius=lineWidth suitable?
> 
> half of the lineWidth I guess

Also, probably only suitable if you want a rounded edge.

**Comment by Woovie** [comment]
I like the idea of it being a useful script, and yeah the love2d options look very nice.

**Comment by qaisjp** [comment]
I propose adding additional arguments to the end of `dxDrawLine3D`. One argument which is a string for the LineJoin mode, and then support an infinite number of subsequent points.

**Old**

`bool dxDrawLine3D ( float startX, float startY, float startZ, float endX, float endY, float endZ [, int color = 0xFFFFFFFF, float width = 1.0, bool postGUI = false ] )`

**New**

`bool dxDrawLine3D ( float x1, float y1, float z1, float x2, float y2, float z2 [, int color = 0xFFFFFFFF, float width = 1.0, bool postGUI = false, string lineJoin = "miter", float x3, float y3, float z3, ... ] )`

And do the same for `dxDrawLine` I guess, but probably as a separate PR/issue.
